src/eko/evolution_operator/__init__.py

```diff
@@ -466,11 +471,18 @@ def integrate(
         tot_start_time = time.perf_counter()
 
         # run integration in parallel for each grid point
-        with Pool(self.n_pools) as pool:
-            res = pool.map(
+        # or avoid opening a single pool
+        if self.n_pools == 1:
```

Comment:
If you wish you can be even stricter and define before `args = (a,b)` and then `map(*args)`